DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -49,23 +56,82 @@ public IActionResult EditDetails()
         }
 
         [HttpPost]
-        public IActionResult EditDetails(EditDetailsViewModel model)
+        public IActionResult EditDetails(EditDetailsViewModel model, string action)
+        {
+            return action switch
+            {
+                "save" => EditDetailsPostSave(model),
+                "previewImage" => EditDetailsPostPreviewImage(model),
+                "removeImage" => EditDetailsPostRemoveImage(model),
+                _ => View(model),
+            };
+        }
+
+        private IActionResult EditDetailsPostSave(EditDetailsViewModel model)
         {
             if (!ModelState.IsValid)
             {
                 return View(model);
             }
 
+            if (model.ProfilePicture != null)
+            {
+                ModelState.AddModelError("ProfilePicture", "Preview your new profile picture before saving");
+                return View(model);
+            }
+
             var userAdminId = User.GetAdminId();
             var userDelegateId = User.GetNullableCandidateId();
 
-            if (!userService.TryUpdateUserAccountDetails(userAdminId, userDelegateId, model.Password,  model.FirstName, model.LastName, model.Email))
+            var profileImageToSave = model.HasProfileImageBeenRemoved ? null : model.ProfileImage;
+
+            if (!userService.TryUpdateUserAccountDetails(userAdminId, userDelegateId, model.Password,  model.FirstName, model.LastName, model.Email, profileImageToSave))
             {
                 ModelState.AddModelError("Password", "The password you have entered is incorrect.");
                 return View(model);
             }
 
             return RedirectToAction("Index");
         }
+
+        private IActionResult EditDetailsPostPreviewImage(EditDetailsViewModel model)
+        {
+            // We don't want to display validation errors on other fields in this case
+            foreach (var key in ModelState.Keys.Where(k => k != "ProfilePicture"))
+            {
+                ModelState[key].Errors.Clear();
+                ModelState[key].ValidationState = ModelValidationState.Valid;
+            }
+
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            if (model.ProfilePicture != null)
+            {
+                ModelState.Remove("NewProfileImage");

```

Comment:
Do we have the "NewProfileImage" field in the model anymore?

Reply:
Ah whoops, problems with renaming and strings not being checked. Will update them all to be using nameof instead a string to prevent doing that again in the future. Might be something worth doing everywhere we are using these string property names (unless there is some reason you know not to use this)